AndroidAssignment/app/src/main/java/org/sopt/data/repository/HomeRepoImpl.kt

```diff
@@ -1,4 +1,59 @@
 package org.sopt.data.repository
 
-class HomeRepoImpl {
+import androidx.lifecycle.LiveData
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.sopt.data.local.dao.ProfileDao
+import org.sopt.data.local.dao.RepoDao
+import org.sopt.data.local.entity.ProfileData
+import org.sopt.data.local.entity.RepoData
+import javax.inject.Inject
+
+class HomeRepoImpl@Inject constructor(private val profileDao: ProfileDao, private val repoDao: RepoDao) : HomeRepo {
+    override fun getProfileDataAll(): LiveData<List<ProfileData>> = profileDao.getAll()
+
+    override fun insertProfileData(profileData: ProfileData) {
+        try {
+            GlobalScope.launch {
+                profileDao.insert(profileData)
+            }
```

Comment:
지난번 리뷰에서도 말했지만 GlobalScope는 그다지 좋지 못한 코루틴 usecase인 것 같음. 차라리 Dispatchers.IO 스레드를 통해서 DB에서 CRUD하는 게 더 좋을 것 같아.
```suggestion
withContext(Dispatchers.IO) {
    // do crud in here
}
```

Reply:
으악... 안그래도 GlobalScope랑 RunBlocking 때문에 여러가지로 알아봤는데.. 코루틴 Flow 방식을 보는데, 너무너무 어려워서 일단 그대로 놔두기는 했거든..? 근데 그 태환님 블로그에도 간단한 작업의 경우는 RunBlocking으로도 한다고 적혀있길래.. 일단 더 알아봐야겠다.. 이번주는 어차피 과제도 없어서..ㅋㅋㅋㅋㅋ